notebooks/working/reproducing_deconfounder_results.ipynb

```diff
@@ -0,0 +1,1680 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['investigating_deconfounder.py',\n",
+       " 'investigating_deconfounder.ipynb',\n",
+       " '.gitkeep',\n",
+       " '__pycache__',\n",
+       " 'factor_models.py',\n",
+       " '.ipynb_checkpoints',\n",
+       " 'reproducing_deconfounder_results.ipynb']"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "from causalgraphicalmodels import CausalGraphicalModel, StructuralCausalModel\n",
+    "import pylogit\n",
+    "from collections import OrderedDict\n",
+    "import pylogit as cm\n",
+    "from functools import reduce\n",
+    "import statsmodels.api as sm\n",
+    "import statsmodels.formula.api as smf\n",
+    "from math import ceil\n",
+    "\n",
+    "from IPython import display\n",
+    "import seaborn as sns\n",
+    "\n",
+    "import numpy as np\n",
+    "import numpy.random as npr\n",
+    "import pandas as pd\n",
+    "import tensorflow as tf\n",
+    "import tensorflow_probability as tfp\n",
+    "import statsmodels.api as sm\n",
+    "\n",
+    "from tensorflow_probability import edward2 as ed\n",
+    "from sklearn.datasets import load_breast_cancer\n",
+    "from pandas.plotting import scatter_matrix\n",
+    "from scipy import sparse, stats\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.linear_model import LogisticRegression \n",
+    "from sklearn.metrics import confusion_matrix, classification_report, roc_auc_score, roc_curve\n",
+    "\n",
+    "import matplotlib\n",
+    "matplotlib.rcParams.update({'font.sans-serif' : 'Helvetica',\n",
+    "                            'axes.labelsize': 10,\n",
+    "                            'xtick.labelsize' : 6,\n",
+    "                            'ytick.labelsize' : 6,\n",
+    "                            'axes.titlesize' : 10})\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "import seaborn as sns\n",
+    "color_names = [\"windows blue\",\n",
+    "               \"amber\",\n",
+    "               \"crimson\",\n",
+    "               \"faded green\",\n",
+    "               \"dusty purple\",\n",
+    "               \"greyish\"]\n",
+    "colors = sns.xkcd_palette(color_names)\n",
+    "sns.set(style=\"white\", palette=sns.xkcd_palette(color_names), color_codes = False)\n",
+    "\n",
+    "from factor_models import *\n",
```

Comment:
Unless there's a good reason / if possible, avoid the wildcard import (i.e. the "*") by explicitly importing the functions that you need. (See [PEP8](https://www.python.org/dev/peps/pep-0008/))